src/Microsoft.Extensions.Configuration/ConfigurationRoot.cs

```diff
@@ -24,6 +24,12 @@ public ConfigurationRoot(IList<IConfigurationProvider> providers)
             _providers = providers;
         }
 
+        public string Key => string.Empty;
+
+        public string Path => string.Empty;
+
+        public string Value { get; set; }
```

Comment:
See my comment above. It turns out that on Configuration empty keys are valid keys and not the same as addressing "this" `IConfiguration`. It probably means there no valid thing `root.Value` can do.